beacon/lib/beacon/partition.ex

```diff
@@ -78,21 +93,15 @@ defmodule Beacon.Partition do
   @spec handle_call({:join, Beacon.group(), pid}, GenServer.from(), State.t()) ::
           {:reply, :ok, State.t()}
   def handle_call({:join, group, pid}, _from, state) do
-    case :ets.lookup(state.name, group) do
-      [{^group, pids, counter}] ->
-        if MapSet.member?(pids, pid) do
-          # Already being tracked
-          {:reply, :ok, state}
-        else
-          new_pids = MapSet.put(pids, pid)
-          :ets.insert(state.name, {group, new_pids, counter + 1})
-          ref = Process.monitor(pid, tag: {:DOWN, group})
-          monitors = Map.put(state.monitors, {group, pid}, ref)
-          {:reply, :ok, %{state | monitors: monitors}}
-        end
+    case :ets.lookup(state.entries_table, {group, pid}) do
```

Comment:
You can use :ets.insert_new, instead of doing a lookup + insert.

Reply:
Oh yup completely forgot about insert_new 🤦